core/src/main/java/org/elasticsearch/common/xcontent/support/AbstractXContentParser.java

```diff
@@ -54,7 +55,31 @@ private static void checkCoerceString(boolean coerce, Class<? extends Number> cl
         }
     }
 
-    private static final DeprecationLogger deprecationLogger = new DeprecationLogger(Loggers.getLogger(AbstractXContentParser.class));
+    // do not use this field directly, use AbstractXContentParser#getDeprecationLogger
+    private static final SetOnce<DeprecationLogger> deprecationLogger = new SetOnce<>();
+
+    private static DeprecationLogger getDeprecationLogger() {
```

Comment:
any reason we didn't use the [holder](https://en.wikipedia.org/wiki/Initialization-on-demand_holder_idiom) idiom here? 

ie:
```Java

private static class Holder {
  static final DeprecationLogger deprecationLogger = new DeprecationLogger(Loggers.getLogger(AbstractXContentParser.class));
}

private static DeprecationLogger getDeprecationLogger() {
  return Holder.deprecationLogger;
}
```